cdap-security/src/main/java/co/cask/cdap/security/server/LDAPAuthenticationHandler.java

```diff
@@ -35,19 +33,14 @@
   private static final List<String> mandatoryConfigurables = ImmutableList.of("debug", "hostname", "port", "userBaseDn",
                                                                               "userRdnAttribute", "userObjectClass");
 
-  private static final List<String> optionalConfigurables = ImmutableList.of("bindDn", "bindPassword", "useLdaps",
+  private static final List<String> optionalConfigurables = ImmutableList.of("bindDn", "useLdaps",
                                                                              "userIdAttribute", "userPasswordAttribute",
                                                                              "roleBaseDn", "roleNameAttribute",
                                                                              "roleMemberAttribute", "roleObjectClass");
-  private static boolean ldapSSLVerifyCertificate = true;
 
-  /**
-   * Create a new Authentication handler to use LDAP for external authentication.
-   */
-  @Inject
-  public LDAPAuthenticationHandler(CConfiguration configuration) throws Exception {
-    super(configuration);
-  }
+  private static final List<String> optionalSecureConfigurables = ImmutableList.of("bindPassword");
```

Comment:
With these changes, `bindPassword` must be configured in cdap-security.xml. Any value in cdap-site.xml will be ignored.
Whether we want to support backwards compatibility is up in the air.

It's not good practice to have that parameter in cdap-site.xml, right? Which is why I think we should force users to move it to cdap-security.xml.